File: DSA_PYTHON/Matrices/SymmetricMatrix.py
class SymmetricMatrix:
    # SymmetricMatrix: A symmetric matrix is a square matrix that is equal to its transpose.
    # approach of code is changed compared to lower,diagonal,upper triangular matrix...
    def read_matrix(self, n):
        matrix = []
        print("Enter the matrix elements row by row:")
        for i in range(n):
            row = []
            for j in range(n):
                val = int(input(f"Element [{i},{j}]: "))
                row.append(val)
            matrix.append(row)
        return matrix

    def print_matrix(self, matrix):
        print("\nMatrix:")
        for row in matrix:
            for val in row:
                print(val, end="\t")
            print()

    def is_symmetric(self, matrix):
        n = len(matrix)
        for i in range(n):
            for j in range(n):
                if matrix[i][j] != matrix[j][i]:
                    return False
        return True

    def main(self):
        n = int(input("Enter the size of the square matrix (n x n): "))
        matrix = self.read_matrix(n)
        self.print_matrix(matrix)

        if self.is_symmetric(matrix):
            print("\n The matrix is symmetric.")
        else:
            print("\n The matrix is NOT symmetric.")

File: DSA_CSHARP/Matrices/SymmetricMatrix.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DSA_CSHARP.Matrices;
internal class SymmetricMatrix
{
    //SymmetricMatrix: A symmetric matrix is a square matrix that is equal to its transpose.
    void ReadMatrix(int[,] matrix)
    {
        int n = matrix.GetLength(0);
        Console.WriteLine("Enter matrix elements:");
        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < n; j++)
            {
                Console.Write($"Element [{i},{j}]: ");
                matrix[i, j] = int.Parse(Console.ReadLine());
            }
        }
    }
    void PrintMatrix(int[,] matrix)
    {
        int n = matrix.GetLength(0);
        Console.WriteLine("\n Matrix:");
        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < n; j++)
            {
                Console.Write(matrix[i, j] + " ");
            }
            Console.WriteLine();
        }
    }
    bool IsSymmetric(int[,] matrix)
    {
        int n = matrix.GetLength(0);
        for (int i = 0; i < n; i++)
        {
            for (int j = 0; j < n; j++)
            {
                if (matrix[i, j] != matrix[j, i])
                    return false;
            }
        }
        return true;
    }


    public string Main()
    {
        Console.Write("Enter size of the square matrix (n x n): ");
        int n = int.Parse(Console.ReadLine());

        int[,] matrix = new int[n, n];
        ReadMatrix(matrix);
        PrintMatrix(matrix);

        if (IsSymmetric(matrix))
            Console.WriteLine("\n The matrix is symmetric.");
        else
            Console.WriteLine("\n The matrix is NOT symmetric.");

        return "";
    }
}
